File: src/S1/BOofLogN/Python/main.py
from binary_search_helper import binary_search

test_array = [3, 6, 9, 12, 15, 18, 21, 24]


def perform_binary_search(array, target):
    result = binary_search(array, target)

    if result != -1:
        print(f"Binary Search: Found target {target} at index {result}")
    else:
        print(f"Binary Search: Target {target} not found in the array.")


# Test with target = 9
perform_binary_search(test_array, 9)

# Test with target = 99
perform_binary_search(test_array, 99)

File: src/S1/BOofLogN/JS/main.js
// main.js

const { binarySearch } = require('./binarySearchHelper');

const testArray = [3, 6, 9, 12, 15, 18, 21, 24];

function performBinarySearch(array, target) {
    const result = binarySearch(array, target);

    if (result !== -1) {
        console.log(`Binary Search: Found target ${target} at index ${result}`);
    } else {
        console.log(`Binary Search: Target ${target} not found in the array.`);
    }
}

// Test with target = 9
performBinarySearch(testArray, 9);

// Test with target = 99
performBinarySearch(testArray, 99);
